delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -590,3 +682,365 @@ def list_ports(self, storage_id):
+                                                           target))
+                metrics.append(metric_value)
+
+    @staticmethod
+    def count_metric_data(last_data, now_data, interval, target, metric_type,
+                          metric_map, res_id):
+        if not target:
+            return
+        if 'CACHEHITRATIO' not in metric_type.upper():
+            value = SSHHandler.count_difference(now_data.get(target),
+                                                last_data.get(target))
+        else:
+            value = now_data.get(
+                SSHHandler.VOLUME_PERF_METRICS.get(metric_type))
+        if 'THROUGHPUT' in metric_type.upper():
+            value = value / interval / units.Mi
+        elif 'IOSIZE' in metric_type.upper():
+            value = value / units.Ki
+        elif 'IOPS' in metric_type.upper() or 'RESPONSETIME' \
+                in metric_type.upper():
+            value = value / interval
+        value = round(value, 2)
+        if metric_map.get(res_id):
+            if metric_map.get(res_id).get(metric_type):
+                if metric_map.get(res_id).get(metric_type).get(
+                        now_data.get('time')):
+                    metric_map[res_id][metric_type][now_data.get('time')] \
+                        += value
+                else:
+                    metric_map[res_id][metric_type][now_data.get('time')] \
+                        = value
+            else:
+                metric_map[res_id][metric_type] = {now_data.get('time'): value}
+        else:
+            metric_map[res_id] = {metric_type: {now_data.get('time'): value}}
+
+    @staticmethod
+    def count_difference(now_value, last_value):
+        value = 0
+        if now_value >= last_value:
+            value = now_value - last_value
+        else:
+            value = now_value
+        return value
+
+    @staticmethod
+    def handle_volume_cach_hit(now_data, last_data):
+        rh = SSHHandler.count_difference(now_data.get('rh'),
+                                         last_data.get('rh'))
+        wh = SSHHandler.count_difference(now_data.get('wh'),
+                                         last_data.get('wh'))
+        rht = SSHHandler.count_difference(now_data.get('rht'),
+                                          last_data.get('rht'))
+        wht = SSHHandler.count_difference(now_data.get('wht'),
+                                          last_data.get('wht'))
+        rhr = rh * 100 / rht if rht > 0 else 0
+        whr = wh * 100 / wht if wht > 0 else 0
+        hrt = rhr + whr
+        now_data['rhr'] = rhr
+        now_data['whr'] = whr
+        now_data['hrt'] = hrt
+
+    def get_date_from_each_file(self, file, metric_map, target_list,
+                                resource_type, last_data):
+        with self.ssh_pool.item() as ssh:
+            local_path = '%s/%s' % (
+                os.path.abspath(os.path.join(os.getcwd())),
+                consts.LOCAL_FILE_PATH)
```

Comment:
There's an extra backslash here.
![Uploading image.png…]()

Reply:
this is an file path,need that